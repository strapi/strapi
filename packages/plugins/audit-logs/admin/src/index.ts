import pluginId from './pluginId';

export default {
  register(app: any) {
    app.registerPlugin({
      id: pluginId,
      name: 'Audit Logs',
    });
  },
  bootstrap(app: any) {},
};

