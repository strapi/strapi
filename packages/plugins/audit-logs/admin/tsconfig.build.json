{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "declarationDir": "../dist/admin/src"
  },
  "exclude": ["node_modules/", "dist/"]
}

