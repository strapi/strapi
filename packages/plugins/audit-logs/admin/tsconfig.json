{
  "extends": "@strapi/typescript-utils/tsconfigs/admin",
  "compilerOptions": {
    "outDir": "../dist",
    "rootDir": "."
  },
  "include": ["src/**/*"],
  "exclude": ["node_modules/", "dist/"]
}

