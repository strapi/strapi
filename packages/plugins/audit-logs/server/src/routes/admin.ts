export default {
  type: 'admin',
  routes: [
    {
      method: 'GET',
      path: '/logs',
      handler: 'audit-logs.find',
      config: {
        policies: ['admin::isAuthenticatedAdmin'],
      },
    },
    {
      method: 'GET',
      path: '/logs/stats',
      handler: 'audit-logs.stats',
      config: {
        policies: [
          'admin::isAuthenticatedAdmin',
          {
            name: 'admin::hasPermissions',
            config: {
              actions: ['plugin::audit-logs.read'],
            },
          },
        ],
      },
    },
    {
      method: 'GET',
      path: '/logs/:id',
      handler: 'audit-logs.findOne',
      config: {
        policies: [
          'admin::isAuthenticatedAdmin',
          {
            name: 'admin::hasPermissions',
            config: {
              actions: ['plugin::audit-logs.read'],
            },
          },
        ],
      },
    },
  ],
};

