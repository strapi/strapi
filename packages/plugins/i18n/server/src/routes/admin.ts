export default {
  type: 'admin',
  routes: [
    {
      method: 'GET',
      path: '/iso-locales',
      handler: 'iso-locales.listIsoLocales',
      config: {
        policies: [
          'admin::isAuthenticatedAdmin',
          {
            name: 'plugin::content-manager.hasPermissions',
            config: { actions: ['plugin::i18n.locale.read'] },
          },
        ],
      },
    },
    {
      method: 'GET',
      path: '/locales',
      handler: 'locales.listLocales',
      config: {
        policies: ['admin::isAuthenticatedAdmin'],
      },
    },
    {
      method: 'POST',
      path: '/locales',
      handler: 'locales.createLocale',
      config: {
        policies: [
          'admin::isAuthenticatedAdmin',
          {
            name: 'plugin::content-manager.hasPermissions',
            config: { actions: ['plugin::i18n.locale.create'] },
          },
        ],
      },
    },
    {
      method: 'PUT',
      path: '/locales/:id',
      handler: 'locales.updateLocale',
      config: {
        policies: [
          'admin::isAuthenticatedAdmin',
          {
            name: 'plugin::content-manager.hasPermissions',
            config: { actions: ['plugin::i18n.locale.update'] },
          },
        ],
      },
    },
    {
      method: 'DELETE',
      path: '/locales/:id',
      handler: 'locales.deleteLocale',
      config: {
        policies: [
          'admin::isAuthenticatedAdmin',
          {
            name: 'plugin::content-manager.hasPermissions',
            config: { actions: ['plugin::i18n.locale.delete'] },
          },
        ],
      },
    },
    {
      method: 'POST',
      path: '/content-manager/actions/get-non-localized-fields',
      handler: 'content-types.getNonLocalizedAttributes',
      config: {
        policies: ['admin::isAuthenticatedAdmin'],
      },
    },
  ],
};
