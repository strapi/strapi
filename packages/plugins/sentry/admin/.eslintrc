{
  "root": true,
  "extends": ["custom/front/typescript"]
}
