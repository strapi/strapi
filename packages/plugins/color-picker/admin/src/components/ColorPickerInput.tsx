import * as React from 'react';

import {
  Button,
  Box,
  Field,
  Flex,
  Popover,
  Typography,
  useComposedRefs,
} from '@strapi/design-system';
import { CaretDown } from '@strapi/icons';
import { type InputProps, useField } from '@strapi/strapi/admin';
import { HexColorPicker } from 'react-colorful';
import { useIntl } from 'react-intl';
import { styled } from 'styled-components';

import { getTrad } from '../utils/getTrad';

const ColorPreview = styled.div`
  border-radius: 50%;
  width: 20px;
  height: 20px;
  margin-right: 10px;
  background-color: ${(props) => props.color};
  border: 1px solid rgba(0, 0, 0, 0.1);
`;

const ColorPicker = styled(HexColorPicker)`
  && {
    width: 100%;
    aspect-ratio: 1.5;
  }

  .react-colorful__pointer {
    width: ${({ theme }) => theme.spaces[3]};
    height: ${({ theme }) => theme.spaces[3]};
  }

  .react-colorful__saturation {
    border-radius: ${({ theme }) => theme.spaces[1]};
    border-bottom: none;
  }

  .react-colorful__hue {
    border-radius: 10px;
    height: ${({ theme }) => theme.spaces[3]};
    margin-top: ${({ theme }) => theme.spaces[2]};
  }
`;

const ColorPickerToggle = styled(Button)`
  & > span {
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 100%;
  }

  svg {
    width: ${({ theme }) => theme.spaces[2]};
    height: ${({ theme }) => theme.spaces[2]};
  }

  svg > path {
    fill: ${({ theme }) => theme.colors.neutral500};
    justify-self: flex-end;
  }
`;

const ColorPickerPopover = styled(Popover.Content)`
  padding: ${({ theme }) => theme.spaces[2]};
  min-height: 270px;
`;

type ColorPickerInputProps = InputProps & {
  labelAction?: React.ReactNode;
};

export const ColorPickerInput = React.forwardRef<HTMLButtonElement, ColorPickerInputProps>(
  ({ hint, disabled, labelAction, label, name, required, ...props }, forwardedRef) => {
    const [showColorPicker, setShowColorPicker] = React.useState(false);
    const colorPickerButtonRef = React.useRef<HTMLButtonElement>(null!);
    const { formatMessage } = useIntl();
    const field = useField(name);
    const color = field.value ?? '#000000';

    const composedRefs = useComposedRefs(forwardedRef, colorPickerButtonRef);

    return (
      <Field.Root name={name} id={name} error={field.error} hint={hint} required={required}>
        <Flex direction="column" alignItems="stretch" gap={1}>
          <Field.Label action={labelAction}>{label}</Field.Label>
          <Popover.Root onOpenChange={setShowColorPicker}>
            <Popover.Trigger>
              <ColorPickerToggle
                ref={composedRefs}
                aria-label={formatMessage({
                  id: getTrad('color-picker.toggle.aria-label'),
                  defaultMessage: 'Color picker toggle',
                })}
                aria-controls="color-picker-value"
                aria-haspopup="dialog"
                aria-expanded={showColorPicker}
                aria-disabled={disabled}
                disabled={disabled}
                variant="tertiary"
                size="L"
              >
                <Flex>
                  <ColorPreview color={color} />
                  <Typography
                    style={{ textTransform: 'uppercase' }}
                    textColor={field.value ? undefined : 'neutral600'}
                    variant="omega"
                  >
                    {color}
                  </Typography>
                </Flex>
                <CaretDown aria-hidden />
              </ColorPickerToggle>
            </Popover.Trigger>
            <ColorPickerPopover sideOffset={4}>
              <ColorPicker color={color} onChange={(hexValue) => field.onChange(name, hexValue)} />
              <Flex paddingTop={3} paddingLeft={4} justifyContent="flex-end">
                <Box paddingRight={2}>
                  <Typography variant="omega" tag="label" textColor="neutral600">
                    {formatMessage({
                      id: getTrad('color-picker.input.format'),
                      defaultMessage: 'HEX',
                    })}
                  </Typography>
                </Box>
                <Field.Root>
                  <Field.Input
                    aria-label={formatMessage({
                      id: getTrad('color-picker.input.aria-label'),
                      defaultMessage: 'Color picker input',
                    })}
                    style={{ textTransform: 'uppercase' }}
                    name={name}
                    defaultValue={color}
                    placeholder="#000000"
                    onChange={field.onChange}
                    {...props}
                  />
                </Field.Root>
              </Flex>
            </ColorPickerPopover>
          </Popover.Root>
          <Field.Hint />
          <Field.Error />
        </Flex>
      </Field.Root>
    );
  }
);
