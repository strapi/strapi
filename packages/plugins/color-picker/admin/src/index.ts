import { ColorPickerIcon } from './components/ColorPickerIcon';
import { pluginId } from './pluginId';
import { getTrad } from './utils/getTrad';
import { prefixPluginTranslations } from './utils/prefixPluginTranslations';

// eslint-disable-next-line import/no-default-export
export default {
  /**
   * TODO: we need to have the type for StrapiApp done from `@strapi/admin` package.
   */
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  register(app: any) {
    app.customFields.register({
      name: 'color',
      pluginId: 'color-picker',
      type: 'string',
      icon: ColorPickerIcon,
      intlLabel: {
        id: getTrad('color-picker.label'),
        defaultMessage: 'Color',
      },
      intlDescription: {
        id: getTrad('color-picker.description'),
        defaultMessage: 'Select any color',
      },
      components: {
        Input: async () =>
          import('./components/ColorPickerInput').then((module) => ({
            default: module.ColorPickerInput,
          })),
      },
      options: {
        advanced: [
          {
            intlLabel: {
              id: getTrad('color-picker.options.advanced.regex'),
              defaultMessage: 'RegExp pattern',
            },
            name: 'regex',
            type: 'text',
            defaultValue: '^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$',
            description: {
              id: getTrad('color-picker.options.advanced.regex.description'),
              defaultMessage: 'The text of the regular expression',
            },
          },
          {
            sectionTitle: {
              id: 'global.settings',
              defaultMessage: 'Settings',
            },
            items: [
              {
                name: 'required',
                type: 'checkbox',
                intlLabel: {
                  id: getTrad('color-picker.options.advanced.requiredField'),
                  defaultMessage: 'Required field',
                },
                description: {
                  id: getTrad('color-picker.options.advanced.requiredField.description'),
                  defaultMessage: "You won't be able to create an entry if this field is empty",
                },
              },
            ],
          },
        ],
      },
    });
  },
  async registerTrads({ locales }: { locales: string[] }) {
    const importedTrads = await Promise.all(
      locales.map((locale) => {
        return import(`./translations/${locale}.json`)
          .then(({ default: data }) => {
            return {
              data: prefixPluginTranslations(data, pluginId),
              locale,
            };
          })
          .catch(() => {
            return {
              data: {},
              locale,
            };
          });
      })
    );

    return Promise.resolve(importedTrads);
  },
};
