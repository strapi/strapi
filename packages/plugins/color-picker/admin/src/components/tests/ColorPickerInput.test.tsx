import { DesignSystemProvider } from '@strapi/design-system';
import { Form } from '@strapi/strapi/admin';
import { render as renderRTL } from '@testing-library/react';
import { userEvent } from '@testing-library/user-event';
import { IntlProvider } from 'react-intl';

import { ColorPickerInput } from '../ColorPickerInput';

const render = () => ({
  ...renderRTL(<ColorPickerInput name="color" label={'color-picker'} type="string" />, {
    wrapper: ({ children }) => {
      const locale = 'en';
      return (
        <IntlProvider locale={locale} messages={{}} textComponent="span">
          <DesignSystemProvider locale={locale}>
            <Form onSubmit={jest.fn()} method="POST">
              {children}
            </Form>
          </DesignSystemProvider>
        </IntlProvider>
      );
    },
  }),
  user: userEvent.setup(),
});

describe('<ColorPickerInput />', () => {
  it('renders and matches the snapshot', () => {
    const { container } = render();

    expect(container).toMatchSnapshot();
  });

  it('toggles the popover', async () => {
    const { user, getByRole } = render();
    await user.click(getByRole('button', { name: 'Color picker toggle' }));

    expect(getByRole('dialog')).toBeVisible();
    expect(getByRole('slider', { name: 'Color' })).toBeVisible();
    expect(getByRole('slider', { name: 'Hue' })).toBeVisible();
    expect(getByRole('textbox', { name: 'Color picker input' })).toBeVisible();
    expect(getByRole('textbox', { name: 'Color picker input' })).toHaveValue('#000000');
  });

  it('can change color via text input', async () => {
    const { user, getByRole } = render();

    await user.click(getByRole('button', { name: 'Color picker toggle' }));

    const textInput = getByRole('textbox', { name: 'Color picker input' });
    const testColor = '#ff3c0c';

    await user.clear(textInput);
    await user.type(textInput, testColor);

    expect(textInput).toHaveValue(testColor);
  });
});
