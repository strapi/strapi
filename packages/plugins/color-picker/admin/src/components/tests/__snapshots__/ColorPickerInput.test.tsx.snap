// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColorPickerInput /> renders and matches the snapshot 1`] = `
.c3 {
  padding-inline-start: 16px;
  padding-inline-end: 16px;
  border-radius: 4px;
  cursor: pointer;
}

.c0 {
  gap: 4px;
  align-items: stretch;
  flex-direction: column;
  display: flex;
}

.c4 {
  gap: 8px;
  align-items: center;
  justify-content: center;
  flex-direction: row;
  display: inline-flex;
}

.c8 {
  align-items: center;
  flex-direction: row;
  display: flex;
}

.c1 {
  font-size: 1.2rem;
  line-height: 1.33;
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  color: #32324d;
  font-weight: 600;
}

.c7 {
  font-size: 1.4rem;
  line-height: 1.43;
  color: currentcolor;
  font-weight: 600;
}

.c10 {
  font-size: 1.4rem;
  line-height: 1.43;
  color: #666687;
}

.c11 {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c5 {
  height: 4rem;
  text-decoration: none;
  border: 1px solid #dcdce4;
  background: #ffffff;
  color: #32324d;
}

.c5:hover {
  background-color: #f6f6f9;
}

.c5:active {
  background-color: #eaeaef;
}

.c5[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
  color: #666687;
  cursor: default;
}

.c2 {
  display: flex;
}

.c9 {
  border-radius: 50%;
  width: 20px;
  height: 20px;
  margin-right: 10px;
  background-color: #000000;
  border: 1px solid rgba(0, 0, 0, 0.1);
}

.c6>span {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
}

.c6 svg {
  width: 8px;
  height: 8px;
}

.c6 svg>path {
  fill: #8e8ea9;
  justify-self: flex-end;
}

@media (prefers-reduced-motion: no-preference) {
  .c5 {
    transition: background-color 120ms cubic-bezier(0.25, 0.46, 0.45, 0.94),color 120ms cubic-bezier(0.25, 0.46, 0.45, 0.94),border-color 200ms cubic-bezier(0.25, 0.46, 0.45, 0.94);
  }
}

<div>
  <form
    class=""
    method="POST"
    novalidate=""
  >
    <div
      class="c0"
    >
      <div
        class="c0"
      >
        <label
          class="c1 c2"
          for="color"
          id="color-label"
        >
          color-picker
        </label>
        <button
          aria-controls="color-picker-value"
          aria-disabled="false"
          aria-expanded="false"
          aria-haspopup="dialog"
          aria-label="Color picker toggle"
          class="c3 c4 c5 c6"
          data-state="closed"
          type="button"
        >
          <span
            class="c7"
          >
            <div
              class="c8"
            >
              <div
                class="c9"
                color="#000000"
              />
              <span
                class="c10"
                style="text-transform: uppercase;"
              >
                #000000
              </span>
            </div>
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="16"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m26.708 12.708-10 10a1 1 0 0 1-1.415 0l-10-10A1 1 0 0 1 6 11h20a1 1 0 0 1 .707 1.707"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </form>
  <span
    class="c11"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </span>
</div>
`;
